### ISSUE TITLE
CA1052 (Static holder types) should ignore static constructors

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-analyzers#1162]
#### Analyzer package

Example: Microsoft.CodeQuality.Analyzers

#### Analyzer

CA1052 (Static holder types)

#### Repro steps

1. Declare a non-static class with only a static constructor

```csharp
public class C16
{
    static C16() { }
}
```

#### Expected behavior

CA1052 should not be reported.

#### Actual behavior

CA1052 is reported.

For all practical purposes, C16 is an empty non-static class, which currently does not report a diagnostic.

➡️ More broadly, CA1052 could be updated to ignore all inaccessible members for its analysis (`private` members plus the static constructor).
